### ISSUE TITLE
Hyperopt parallelization issue

### ISSUE THREAD

**Issue by RuiSiang** [freqtrade/freqtrade#8894]
## Describe your environment

  * Operating system: Ubuntu 22.04
  * Python Version: N/A
  * CCXT version: N/A
  * Freqtrade Version: freqtrade 2023.6 (docker, stable)
  
## Your question
I've been testing around hyperopt functionality, got it running on a 16-core server, but found out that the workers (32 with -1 in worker param) are only fully utilized in occasional surges (see screenshot below). The surges happen shortly before the epoch ticker increases. I'm wondering if this is expected, and whether or not there is a workaround because this introduces a bottleneck where machines with a lot more computational power gains much less advantage. Thanks!

![image](https://github.com/freqtrade/freqtrade/assets/49300088/2aabca89-7dee-4945-8328-581a97df21aa)

**Comment by froggleston** [comment]
This is expected. There are single thread elements of hyperopting (data loading), and multiple thread elements (the actual backtesting).

**Comment by RuiSiang** [comment]
Data loading should only occur on startup though. I'm guessing the wait time is mostly `populate_indicators()` that's clogging the performance.

**Comment by xmatthias** [comment]
Neither of the options above.
Dataloading happens only at startup - and is single-process, but is not the problem, really.

the actual finding of the next best points to try (respecting all prior results) is what's the real bottleneck (https://github.com/scikit-optimize/scikit-optimize/issues/768).

Backtesting runs in however many processes you specify (`-j` - or "available -1").
The scikit-optimize algorithm itself is what's the "single core" performance.

Now backtesting runtimes will differ, too (if a bot makes 1000 trades vs. 2 trades makes a huge difference for backtesting) - so parallelism will decline over time within each epoch - reaching 1 when the Bayesian Optimization runs - and will scale out to X again (repeating this process).

In short - not in our control - but in scikit-optimize - or scikit - (don't remember which one has the actual implementation).
